src/main/java/auth/validator/UserValidator.java

```diff
@@ -0,0 +1,51 @@
+package auth.validator;
+
+
+import org.apache.commons.validator.routines.EmailValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+import org.springframework.validation.Validator;
+
+import auth.model.User;
+import auth.service.UserService;
+
+@Component
+public class UserValidator implements Validator {
+
+
+	
+	
+	@Autowired
+	private UserService userService;
+
+	@Override
+	public boolean supports(Class<?> clazz) {
+		return User.class.equals(clazz);
+	}
+
+	@Override
+	public void validate(Object target, Errors errors) {
+		User user = (User) target;
+
+		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "email", "NotEmpty");
+		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "password", "NotEmpty");
+		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "passwordConfirm", "NotEmpty");
+		
+		
+		
+		if (userService.findUserByEmail(user.getEmail()) != null) {
```

Comment:
This validation is hitting database. It should be executed only if required (all other validation passed for target field).